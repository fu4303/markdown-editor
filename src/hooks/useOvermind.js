/* eslint-disable no-param-reassign */
import { createHook } from "overmind-react"

const initialState =
  "# Hello, World!\n\nWelcome to **Markdown Forge**, an online editor for creating and saving markdown files.\n\n _Created by [@JustinJunoDev](https://twitter.com/justinjunodev)._"

const resetState = "So fresh and so clean . . ."

export const config = {
  state: {
    markdown: initialState,
  },
  actions: {
    resetMarkdown({ state }) {
      state.markdown = resetState
    },
    updateMarkdown({ state }, event) {
      state.markdown = event.target.value
    },
    saveMarkdown({ state }) {
      const source = document.createElement("a")
      const file = new Blob([state.markdown], { type: "text/plain" })
      source.href = URL.createObjectURL(file)
      source.download = "markdownForge.md"
      source.click()
    },
  },
}

export const useOvermind = createHook()
