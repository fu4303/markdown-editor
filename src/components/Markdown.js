import React from "react"
import ReactMarkdown from "react-markdown"
import { useOvermind } from "../hooks/useOvermind"

const Markdown = () => {
  const {
    state,
    actions: { resetMarkdown, updateMarkdown, saveMarkdown },
  } = useOvermind()

  return (
    <>
      <textarea onChange={(e) => updateMarkdown(e)} value={state.markdown} />
      <ReactMarkdown source={state.markdown} />
      <button type="button" onClick={() => resetMarkdown()}>
        Test Reset
      </button>
      <button type="button" onClick={() => saveMarkdown()}>
        Test Save
      </button>
    </>
  )
}

export default Markdown
