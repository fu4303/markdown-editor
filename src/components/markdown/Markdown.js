import React from "react"
import Fade from "react-reveal/Fade"
import { useOvermind } from "../../hooks/useOvermind"
import { MarkdownWrapper, MarkdownEditor, MarkdownPreview } from "./styles"

const Markdown = () => {
  const {
    state,
    actions: { updateMarkdown },
  } = useOvermind()

  return (
    <Fade delay={250} duration={1500}>
      <MarkdownWrapper>
        <MarkdownEditor
          onChange={(e) => updateMarkdown(e)}
          value={state.markdown}
        />
        <MarkdownPreview source={state.markdown} />
      </MarkdownWrapper>
    </Fade>
  )
}

export default Markdown
