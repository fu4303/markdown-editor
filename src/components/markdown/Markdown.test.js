import React from "react"
import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import { createOvermindMock } from "overmind"
import { Provider } from "overmind-react"
import { config } from "../../hooks/useOvermind"
import Markdown from "./Markdown"

const mockedOvermind = createOvermindMock(config)

describe("<Markdown />", () => {
  it("renders without crashing", () => {
    shallow(
      <Provider value={mockedOvermind}>
        <Markdown />
      </Provider>
    )
  })

  it("matches snapshot with context passed correctly", () => {
    const wrapper = shallow(
      <Provider value={mockedOvermind}>
        <Markdown />
      </Provider>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
