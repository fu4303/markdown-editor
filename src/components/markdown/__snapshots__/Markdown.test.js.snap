// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Markdown /> matches snapshot with context passed correctly 1`] = `<Markdown />`;
